### PR TITLE
luba.tarasenko

### PR DESCRIPTION
**Pull request by lubo4kaTarasenko**

### CODE REVIEW

**Review by slenderock — APPROVED** [review]
great job. love it

File: liubov.tarasenko/homework2/task2_2/parenthesis.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+end 
+puts '<<}'.valid?
+puts '<<}>'.valid?
+puts '<<}>>'.valid?
+puts '<<{}>>'.valid? 
+puts '<([{})]>>'.valid?
+puts '<<{([])}>>'.valid?
+puts ')('.valid?
+puts '({}){}'.valid?
```
Comment (slenderock) [inline liubov.tarasenko/homework2/task2_2/parenthesis.rb]: Next time please make sure u have a newline at end of file

https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

Comment (slenderock) [reply liubov.tarasenko/homework2/task2_2/parenthesis.rb]: @lubo4kaTarasenko

File: liubov.tarasenko/homework3/prime.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    #prime_matrix = Array.new(n) { Array.new( @all) }
+    prime_matrix = Array.new(n) { @all; @all = @all.rotate(1) }
+    prime_matrix = prime_matrix.rotate(-1)
+    prime_matrix.map{|el| el.join(" ")}
+  end
+
+end
+
+m = PrimeMatrix.new(100)
+m.print_arr
```
Comment (lubo4kaTarasenko) [inline liubov.tarasenko/homework3/prime.rb:23]: prime

File: liubov.tarasenko/homework3/prime.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+class PrimeMatrix
+  def initialize(max)
+    @max = max
+  end
+  
+  def prime_arr
+    base = [2,3,5,7].select { |x| x <= @max }    
+    rest = (2..@max).to_a.reject { |x| base.any? { |b| x % b == 0 } }
```
Comment (slenderock) [inline liubov.tarasenko/homework3/prime.rb]:
```suggestion
    rest = (2..@max).reject { |x| base.any? { |b| x % b == 0 } }
```

Usage of `to_a` is redundant here. 
`Range` class includes almost all `Enumerable` actions.

File: liubov.tarasenko/homework3/prime.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  def print_arr
+    puts calculate
+  end
+
+  def calculate
+    all = prime_arr
+    prime_matrix = Array.new(all.size) { |i| all.rotate(i) }
+    prime_matrix.map { |el| el.join(" ") }
```
Comment (slenderock) [inline liubov.tarasenko/homework3/prime.rb]: Let's have display logic separated from the calculation logic. 
Please move all display transformation to the print_arr method.
`calculate` should return data in the proper correct format, without any serialization

File: liubov.tarasenko/homework3/prime.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
```
Comment (slenderock) [inline liubov.tarasenko/homework3/prime.rb]: an extra newline is not needed here.

**Review by slenderock — APPROVED** [review]

File: liubov.tarasenko/homework3/prime.rb (hunk 4 above)
Comment (lubo4kaTarasenko) [reply liubov.tarasenko/homework3/prime.rb]: thank you for review. just fixed it)

**Pull request merged**